File: 2021/2/2.rb
def p1(command_pairs)
  horizontal = 0
  depth = 0

  command_pairs.each do |command, step|
    case command
    when "forward" then horizontal += step
    when "up" then depth -= step
    when "down" then depth += step
    end
  end
  
  puts "Problem 1: #{horizontal * depth}"
end

def p2(command_pairs)
  horizontal = 0
  depth = 0
  aim = 0

  command_pairs.each do |command, step|
    case command
    when "forward" then 
      horizontal += step
      depth += aim * step
    when "up" then aim -= step
    when "down" then aim += step
    end
  end
  
  puts "Problem 2: #{horizontal * depth}"
end

input = File.open("input").readlines.map(&:chomp)
command_pairs = input.map(&:split).map { |command, step| [command, step.to_i] }

p1(command_pairs)
p2(command_pairs)

File: 2022/4/2.py
file = open("input", "r")
input = list(map(lambda s: s.rstrip('\n'), file.readlines()))
file.close()

any_overlaps = 0
for line in input:
    pair = list(map(lambda s: list(map(lambda i: int(i), s.split("-"))), line.split(",")))
    if pair[0][0] <= pair[1][1] and pair[0][1] >= pair[1][0]:
        any_overlaps += 1

print(any_overlaps)
